File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>
using namespace std;

int main() {
	stack<int> s;      // 스택
	int N;             // 명령의 수
	string command;    // 명령어
	int num;           // push 연산 시, 스택에 넣을 정수
	int result = 0;    // 각 명령어의 결과값

	cin >> N;

	for (int i = 0; i < N; i++) {
		cin >> command;

		// 1. push
		if (command == "push") {
			cin >> num;
			s.push(num);
		}

		// 2. pop
		//    스택이 비어있으면 -1을 출력, 그 외에는 top 값을 출력 후 pop
		else if (command == "pop") {
			if (s.size() == 0) {
				result = -1;
				cout << result << endl;
			}
			else {
				result = s.top();
				cout << result << endl;
				s.pop();
			}
		}

		// 3. size
		else if (command == "size") {
			cout << s.size() << endl;
		}

		// 4. empty
		//    size함수를 통해, size가 0이면 스택이 빈 것이므로 1, 아니면 0 출력     
		else if (command == "empty") {
			if (s.size() == 0) {
				result = 1;
				cout << result << endl;
			}
			else {
				result = 0;
				cout << result << endl;
			}
		}

		// 5. top
		//    스택이 비어있으면 -1을 출력, 그 외에는 top 값을 출력
		else if (command == "top") {
			if (s.size() == 0) {
				result = -1;
				cout << result << endl;
			}
			else {
				result = s.top();
				cout << result << endl;
			}
		}
	}

	return 0;
}

File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

stack = []
for _ in range(int(input())):
    x = input().split()
    if x[0] == 'push':
        stack.append(int(x[1]))
    elif x[0] == 'top':
        if len(stack) == 0 :
            print(-1)
        else :
            print(stack[-1])
    elif x[0] == 'size':
        print(len(stack))
    elif x[0] == 'pop':
        if len(stack) == 0 :
            print(-1)
        else :
            print(stack.pop())
    elif x[0] == 'empty':
        if len(stack) == 0 :
            print(1)
        else :
            print(0)
    
    